pkgs/by-name/xb/xbar/package.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchurl, undmg, ... }:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "xbar";
+  version = "2.1.7-beta";
+  src = fetchurl {
+    url = "https://github.com/matryer/xbar/releases/download/v${finalAttrs.version}/xbar.v${finalAttrs.version}.dmg";
+    sha256 = "sha256-Cn6nxA5NTi7M4NrjycN3PUWd31r4Z0T3DES5+ZAbxz8=";
```

Comment:
Afaik this is undesirable for nixpkgs, particularly for open source software. xbar is open source and released under the MIT license, I would assume we want to compile it from source here.

Reply:
Yes compiling from source is always preferable